### PR TITLE
Add AuditClient for communicating to the Linux kernel

### PR DESCRIPTION
**Pull request by andrewkroh**
Added `AuditClient` for communicating to the Linux kernel's audit interface
via netlink.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ os:
   - osx
 
 go:
-  - 1.7.4
+  - 1.8
```
Comment (ruflin) [inline .travis.yml]: Sounds like we should update the rest too ;-)

File: examples/audit/audit.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+		return errors.Wrap(err, "failed to set audit PID")
+	}
+
+	for {
+		m, err := client.Receive(false)
+		if err != nil {
+			return errors.Wrap(err, "receive failed")
+		}
+
+		if m.MessageType < 1300 || m.MessageType >= 2100 {
```
Comment (ruflin) [inline examples/audit/audit.go]: Should we have some constants for these values to make it more obvious what they mean?

File: examples/audit/audit.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+			continue
+		}
+
+		// Ignore AUDIT_EOE.
+		if m.MessageType == 1320 {
```
Comment (ruflin) [inline examples/audit/audit.go]: Same here, but here the comment helps.

File: sys/linux/audit.go [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+
+	// Retry the non-blocking read multiple times until a response is received.
+	for i := 0; i < 10; i++ {
+		msgs, err = c.netlink.Receive(true, syscall.ParseNetlinkMessage)
+		if err != nil {
+			switch err {
+			case syscall.EINTR:
+				continue
+			case syscall.EAGAIN:
+				time.Sleep(50 * time.Millisecond)
```
Comment (ruflin) [inline sys/linux/audit.go]: That means in total in can take up to 500ms before we get a reply or error?

File: examples/audit/audit.go (hunk 2 above)
Comment (andrewkroh) [inline examples/audit/audit.go]: There will be constants in a future PR. I'm [working an a PR](https://github.com/elastic/gosigar/blob/4393d55dc06805da8051b4907d493daa1e8b53fa/sys/linux/auparse/audit_msg_types.go#L86-L86) to enrich the data and this includes mapping message types to names (i.e 1300 -> "SYSCALL").

File: sys/linux/audit.go (hunk 4 above)
Comment (andrewkroh) [inline sys/linux/audit.go]: Yeah, the communication between user-space and the kernel is asynchronous. So it checks for a response multiple times then finally fails after 500ms.

**Pull request merged**